3711/1/parser/repo.rb

```diff
@@ -0,0 +1,82 @@
+require_relative '../ruby_gem'
+require 'json'
+require 'net/http'
+require 'nokogiri'
+require 'open-uri'
+
+module Parser
+  class Repo
+    CLIENT_ID = '26b9a6fc285e836c9f7a'.freeze
+    CLIENT_SECRET = '7ca0b9c0e86047afbac0951d3d505bf7801a5603'.freeze
+
+    def initialize(name)
+      @name = name
+      @data = {}
+    end
+
+    def parse
+      search_gem && parse_gem_page
+    end
+
+    private
+
+    def search_gem
+      search_params = { q: "#{@name}+language:ruby", sort: 'stars' }
```

Comment:
Shouldn't it be initialized in `def initialize?`

Reply:
I don't think so
As for me it's not a good idea to send some requests for part of data in `initialize` method and then to send other requests for rest of data in `parse` method
It's better to collect all the data in one action